src/main/java/io/minio/MinioClient.java

```diff
@@ -382,26 +382,6 @@ public MinioClient(String endpoint, int port, String accessKey, String secretKey
       throw new InvalidEndpointException("(null)", "null endpoint");
     }
 
-    // for valid URL endpoint, port and secure are ignored
-    HttpUrl url = HttpUrl.parse(endpoint);
```

Comment:
Ah, not sure, but at least all tests pass nicely. Here comes why i dropped the whole block: My first try was to just change the port number in the HttpUrl object, but, that field is final, so no chance there. So what i had to do in case the parameter port and the port in the HttpUrl object did not map, was to use the code following in the lines after (the case where url is null), what left the creation of the url object a kind of useless.